From 791ecb805237a28a038bb317b9894113654f080e Mon Sep 17 00:00:00 2001
From: Alexander Tsoy <alexander@tsoy.me>
Date: Wed, 26 Feb 2014 15:50:17 +0400
Subject: [PATCH 8/8] Add legacy flag (-l) to lz4 and update magic number

Linux kernel does not support the new default lz4 format.

https://bugs.gentoo.org/show_bug.cgi?id=502102
---
 dracut.sh   | 2 +-
 lsinitrd.sh | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/dracut.sh b/dracut.sh
index 5508050..91e05e3 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -782,7 +782,7 @@ case $compress in
     xz)    compress="xz --check=crc32 --lzma2=dict=1MiB -T0";;
     gzip)  compress="gzip -9"; command -v pigz > /dev/null 2>&1 && compress="pigz -9";;
     lzo)   compress="lzop -9";;
-    lz4)   compress="lz4 -9";;
+    lz4)   compress="lz4 -l -9";;
 esac
 if [[ $_no_compress_l = "cat" ]]; then
     compress="cat"
diff --git a/lsinitrd.sh b/lsinitrd.sh
index 2c58f84..8dc9032 100755
--- a/lsinitrd.sh
+++ b/lsinitrd.sh
@@ -172,7 +172,7 @@ case $bin in
     $'\x71\xc7'*|070701)
         CAT="cat --"
         ;;
-    $'\x04\x22'*)
+    $'\x02\x21'*)
         CAT="lz4 -d -c";;
     *)
         CAT="xzcat --";
-- 
1.9.0

